app/code/Magento/Catalog/Block/Product/ListProduct.php

```diff
@@ -233,10 +302,10 @@ protected function _getConfig()
     /**
      * Prepare Sort By fields from Category Data
      *
-     * @param \Magento\Catalog\Model\Category $category
-     * @return \Magento\Catalog\Block\Product\ListProduct
+     * @param Category $category
+     * @return $this
      */
-    public function prepareSortableFieldsByCategory($category)
+    public function prepareSortableFieldsByCategory(Category $category)
```

Comment:
Method signature is changed. This change breaks backward compatibility. Please, revert this.

Reply:
I will revert this change. I added this change because  if $category is not an Category object, it will crash at calling this code  $category->getAvailableSortByOptions() if there is no available orders set on.